hivemind/p2p/p2p_daemon.py

```diff
@@ -402,16 +406,12 @@ async def add_protobuf_handler(
                              (not just ``TInputProtobuf``) as input.
         """
 
-        async def _stream_handler(requests: P2P.TInputStream, context: P2PContext) -> P2P.TOutputStream:
-            if stream_input:
-                input = requests
-            else:
-                count = 0
-                async for input in requests:
-                    count += 1
-                if count != 1:
-                    raise ValueError(f"Got {count} requests for handler {name} instead of one")
+        if not stream_input:
+            await self._add_protobuf_unary_handler(name, handler, input_protobuf_type)
+            return
```

Comment:
This wouldn't work for unary -> stream handlers (we have them in the averager).

A way to fix this is to add `stream_output: bool` parameter to this function (we can set it in `Servicer`) and call the optimized unary handler implementation if `not stream_input and not stream_output`.